### PR TITLE
Simplified the creation of Subclasses of GraphBasedCallGraph

### PR DESCRIPTION
**Pull request by JonasKlauke**
Potential Subclasses gained access to edges in the call graph.
Added printCaller and printCallee to simplify the adpation of the toString method
Added toDotEdge to simplify the export of a call in the dot file generation
Introduced optionals in the graph data handling.

### CODE REVIEW

**Review by swissiety — COMMENTED** [review]
seems convenient - but does this possibly enable a developer to feel confident in a way that the analysis will not fail in cases it should? e.g. thinking that it worked because there is no error and just an empty collection is returned therefore no fact exists.

**Comment by JonasKlauke** [comment]
Are you saying i should print a warning?
I wasnt sure about it, it could happen often

**Comment by swissiety** [comment]
I'm not happy about the warnings either :D Because their are a hint but are often times ignored anyway if its not clear why they are shown or as long as there is no supervisor who complains about them.

We have a similar situation in the `ViewTypeHierarchy` - there we have to check if the TH contains the vertex/`ClassType` - and sometimes we already know they do exist because we got the vertex from the `TypeHierarchy` and in those cases we don't need the additional check. And if its not there we throw an exception so the caller can differentiate between: vertex is unknown and vertex has none of the desired properties the method should calculate (i.e. Collections.empty() would be returned in both cases otherwise). Please check if the analogy is applicable/reasonable in this use case.

**Comment by swissiety** [comment]
TLDR: is it avoidable to run into theses situations (where a log would (then) be printed)?

File: docs/analysisinput.md [hunk 1]
```diff
@@ -117,42 +117,6 @@ JavaView view = new JavaView(inputLocation);
...
-        String outputFile = outDir + apkPath.substring(start + 1, end) + ".jar";
-        Dex2jarCmd.main("-f", apkPath, "-o", outputFile);
-        return outputFile;
-    }
-}
-```
-
-!!! info "A SootUp solution to directly generate Jimple is WIP!"
-
-
```
Comment (swissiety) [inline docs/analysisinput.md]: - [ ] lets keep that code
- [ ] update the text below 
- [ ] add the needed dependency (https://github.com/soot-oss/SootUp/pull/1072)
```maven
		</dependency>
		<dependency>
			<groupId>de.femtopedia.dex2jar</groupId>
			<artifactId>dex2jar</artifactId>
			<version>2.4.22</version>
		</dependency>
```

**Review by swissiety — APPROVED** [review]
nice that this convenient and problem avoiding way was possible! 
please justreintroduce the doc+update ;)

**Review by swissiety — CHANGES_REQUESTED** [review]
nice that this convenient and problem avoiding way was possible! 
please justreintroduce the doc+update ;)

File: docs/analysisinput.md [hunk 2]
```diff
@@ -115,12 +115,12 @@ Path path = Paths.get("Banana.apk");
...
 ```
-
-
 ### Android Bytecode with Dex2Jar
 File-Extensions: `.apk`
 
-The `ApkAnalysisInputLocation` currently uses dex2jar internally 
+If you prefer to use dex2jar as a base to transform android apps to jimple, you can add the code below to create your own analysis input location.
+We used the dependency de.femtopedia.dex2jar:dex2jar:2.4.22 in the given example.
+We recommend to use ApkAnalysisInputLocation
```
Comment (swissiety) [inline docs/analysisinput.md:123]:
```suggestion
We recommend to use `ApkAnalysisInputLocation`

```
	</dependency>
		<dependency>
			<groupId>de.femtopedia.dex2jar</groupId>
			<artifactId>dex2jar</artifactId>
			<version>2.4.22</version>
		</dependency>
```

```

**Review by swissiety — APPROVED** [review]
